src/PolyhedralGeometry/SubdivisionOfPoints/properties.jl

```diff
@@ -4,6 +4,59 @@
 ###############################################################################
 ###############################################################################
 
+#TODO:Code as a SubObjectIterator
+#  so we can apply point_matrix(points(SOP))
+@doc Markdown.doc"""
+    points(SOP::SubdivisionOfPoints)
+
+Return the points of the subdivision of points, `SOP`.
+
+# Examples
+Display the points of the "mother of all examples" non-regular triangulation.
+```jldoctest
+julia> moaepts = [4 0 0; 0 4 0; 0 0 4; 2 1 1; 1 2 1; 1 1 2];
+
+julia> moaeimnonreg0 = IncidenceMatrix([[4,5,6],[1,4,2],[2,4,5],[2,3,5],[3,5,6],[1,3,6],[1,4,6]]);
+
+julia> MOAE = SubdivisionOfPoints(moaepts, moaeimnonreg0);
+
+julia> for p in points(MOAE)
+       println(p)
+       end
+pm::Vector<pm::Rational>
+4 0 0
+pm::Vector<pm::Rational>
+0 4 0
+pm::Vector<pm::Rational>
+0 0 4
+pm::Vector<pm::Rational>
+2 1 1
+pm::Vector<pm::Rational>
+1 2 1
+pm::Vector<pm::Rational>
+1 1 2
+"""
+function points(SOP::SubdivisionOfPoints)
+   PointSOPIterator(SOP)
+end
+
+struct PointSOPIterator
+    SOP::SubdivisionOfPoints
+end
+
+function Base.iterate(iter::PointSOPIterator, index = 1)
+    n_points = npoints(iter.SOP)
+    if index > n_points
+        return nothing
+    end
+    #dehomogenize doesn't check if the first entry is 0. I think this okay here.
+    current_point = dehomogenize(pm_object(iter.SOP).POINTS[index,:])
+    return (current_point, index + 1)
+end
+Base.length(iter::PointSOPIterator) = npoints(iter.SOP)
```

Comment:
Please use @alexej-jordan s iterators here. For example look at how `vertices` constructs such a `SubObjectIterator`. This results in less code duplication and furthermore these iterators can then be used in such things as `point_matrix`, etc..

Reply:
Yes. That is definitely on the TODO. I got stuck on my first attempt on this, and so copied how `maximal_cells(SOP)` was handled. 
Should `maximal_cells(SOP)` also be a SubObjectIterator?

Reply:
Probably. When I started this the iterators weren't in the format they are now. Do you need these as polytohedra?

Reply:
I personally do not at the moment. But somebody certainly in the future would, probably.